qiskit_nature/components/variational_forms/adaptive_ansatz.py

```diff
@@ -0,0 +1,42 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""
+TODO.
+"""
+
+from abc import ABC, abstractmethod
+
+import logging
+
+from qiskit.circuit.library import BlueprintCircuit
+
+logger = logging.getLogger(__name__)
+
+
+class AdaptiveAnsatz(BlueprintCircuit, ABC):
```

Comment:
Do you think we need a separate class for this. I was thinking, since the evolved operator class takes a set of operators that these could be programmatically swapped in and out via the base class. Of course some algos may never use that function, but by having it there it would be available if needed.

Reply:
It is true that the interface of the `EvolvedOperatorAnsatz` currently takes a list of operators. However, I don't think that it makes sense that this ansatz also has about adaptiv behavior (single-responsibility). By splitting this into a separate class/interface, we can also reuse this concept outside of the evolved operators.
@Cryoris what do you think?

Reply:
A class that is responsible for managing a set operators as an ansatz seems fine to me as an encapsulated functionality. Do you have an example of where controlling the set of active operators seems like it would be used elsewhere. While I can see the function could be split out, it seemed to me it could be part of the core function of having/managing a set of operators to evolve as an ansatz. I guess we can see what things look as the design/implementation progresses

Reply:
It would probably be very easy to implement the adaptive behaviour only for the EvolvedOperatorAnsatz, since it already keeps a list of operators and is already able to rebuild itself when the list changes. If we only want this adaptive behaviour for the EvolvedOperatorAnsatz I think we could add it there. We could also add a intermediate `AdaptiveEvolvedOperatorAnsatz` to stick to the single-responsibility design.

If we want a more general adaptive behaviour though, an additional base class seems sensible to me.